module/VuFind/src/VuFind/Crypt/SecretCalculator.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/**
+ * Secret calculator
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2024.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Crypt
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+
+namespace VuFind\Crypt;
+
+use VuFind\Db\Entity\SearchEntityInterface;
+
+/**
+ * Secret calculator
+ *
+ * @category VuFind
+ * @package  Crypt
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class SecretCalculator
```

Comment:
I don't love this class, but the search unsubscribe calculation had to go into a shared location, and it seemed better to create a new class than to overload the existing HMAC class, which seemed like inappropriate scope creep.